### PR TITLE
Monstrous weapons and monster icons

### PR DESCRIPTION
**Pull request by V-ktor**
New graphics for Cosmic Dragon, Juggernaut, Kraken, Krill Swarm, Asteroid Snail and Snowflake. Slot positions are updated as well.
New monstrous weapons and weapon graphics: jaw, claw, tentacle, flame thrower (for dragon), krill fighters (hangar and bay), laser for Snowflakes.

Krill swarms have fighters and fighter bays: 4, 8, 12 fighters are launched per round for medium swarm, large swarm and plague of Krill respectively.

The damage of Juggernauts, Kraken and Snowflakes (medium and large) is increased over time (similar to increased structure of organic hulls).

Added names and descriptions for new weapons to english string table file.

### CODE REVIEW

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -9852,6 +9862,32 @@ Spinal Antimatter Cannon
...
+SR_FLAME_THROWER
+Flame Thrower
+SR_FLAME_THROWER_DESC
+Flame thrower of a space monster, capable of melting most armor.
+
+SR_SNOWFLAKE_LASER
+Laser
+
+SR_SNOWFLAKE_LASER_DESC
+Large Lasers, integrated into the body of a space monster.
```
Comment (Cjkjvfnby) [inline default/stringtables/en.txt:9889]: `Large lasers` ?

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -9852,6 +9862,32 @@ Spinal Antimatter Cannon
...
+
+SR_TENTACLE
+Tentacle
+SR_TENTACLE_DESC
+Tentacle of a Kraken, can damage ships.
+
+SR_FLAME_THROWER
+Flame Thrower
+SR_FLAME_THROWER_DESC
+Flame thrower of a space monster, capable of melting most armor.
```
Comment (Cjkjvfnby) [inline default/stringtables/en.txt]: `Flamethrower` ?

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -9852,6 +9862,32 @@ Spinal Antimatter Cannon
...
+
+SR_CLAWS
+Claws
+SR_CLAWS_DESC
+Claws of a space monster, a powerfull strike of the space monster can deal big damage to a ship.
+
+SR_TENTACLE
+Tentacle
+SR_TENTACLE_DESC
+Tentacle of a Kraken, can damage ships.
```
Comment (Cjkjvfnby) [inline default/stringtables/en.txt]: Should `kraken` be in lowercase?  Like `a dog`

File: default/stringtables/en.txt (hunk 1 above)
Comment (MatGB) [inline default/stringtables/en.txt:9889]: Laser-like attacks? Not sure I like specifically calling them lasers given that's a refined optic effect, etc.

**Comment by MatGB** [comment]
Huge amounts of work here, looks good on first pass, need to test it but thank you for what's there. A couple refinements needed, agree with @Cjkjvfnby on the grammatical points.

File: default/stringtables/en.txt (hunk 2 above)
Comment (geoffthemedio) [inline default/stringtables/en.txt]: How about "Plasma Discharge"?

File: default/stringtables/en.txt (hunk 3 above)
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Instead of "teeth" and "claws", perhaps "spines"?

**Comment by MatGB** [comment]
(also worth noting, better to create a branch for pull requests as sometimes they can sit for awhile before we can get to them, especially if they're big, and that stops you doing other stuff in your master branch on github)

File: default/stringtables/en.txt (hunk 1 above)
Comment (V-ktor) [inline default/stringtables/en.txt:9889]: I don't have a good idea how to call it so the current name is meant to be a placeholder.

Comment (MatGB) [inline default/stringtables/en.txt:9889]: Makes sense. Rename the part SR_SNOWFLAKE_WEAPON and, as a better placeholder, um, "Ice Beams"? Yeah, that's not much better actually (but at least it makes some sense thematically).

File: default/stringtables/en.txt [hunk 4]
```diff
@@ -9852,6 +9862,32 @@ Spinal Antimatter Cannon
...
+Spines
+SR_CLAWS_DESC
+Spines of a space monster, a powerfull strike of the space monster can deal big damage to a ship.
+
+SR_TENTACLE
+Tentacle
+SR_TENTACLE_DESC
+Tentacle of a kraken, can damage ships.
+
+SR_FLAME_THROWER
```
Comment (Cjkjvfnby) [inline default/stringtables/en.txt]: Please rename tokens too, they should look similar to its translated names to avoid confusion.

**Comment by geoffthemedio** [comment]
Fangs sounds better than teeth, I think

**Comment by Cjkjvfnby** [comment]
How about jaws?

File: default/scripting/monster_designs/SM_KRILL_4.focs.txt [hunk 5]
```diff
@@ -3,10 +3,11 @@ ShipDesign
     description = "SM_KRILL_4_DESC"
     hull = "SH_KRILL_4_BODY"
     parts = [
+        "FT_KRILL"
+        "FT_BAY_KRILL"
+        "FT_BAY_KRILL"
+		"FT_BAY_KRILL"
```
Comment (Morlic-fo) [inline default/scripting/monster_designs/SM_KRILL_4.focs.txt]: whitespace

**Comment by Morlic-fo** [comment]
> Krill swarms have fighters and fighter bays: 4, 8, 12 fighters are launched per round for medium swarm, large swarm and plague of Krill respectively.

Keep in mind that the AI doesn't understand fighters yet. Hopefully, I will finish the work on that until the end of month. Probably shouldn't be merged before then.

**Comment by MatGB** [comment]
I would like as much of this as possible in as soon as possible, especially the art. However, we really can't put in the krill as fighters thing (although that is a genius idea and I want it moving forward).  Can we separate that specific thing out (commenting it or similar and leaving standard weapon placeholders?) 

Good work, I especially like the art for the drone factory and the maintenance ship.

**Comment by V-ktor** [comment]
Done. I used jaws as placeholders instead.

As the generic_hull graphic is not used now it could be used as icon for wardens.

File: default/scripting/ship_parts/Weapons.focs.txt [hunk 6]
```diff
@@ -54,4 +54,59 @@ Part
...
+    damage = 5
+    mountableSlotTypes = External
+    buildcost = 30 * [[FLEET_UPKEEP_MULTIPLICATOR]]
+    buildtime = 2
+    location = OwnedBy empire = Source.Owner
+    icon = "icons/ship_parts/tentacle.png"
+
+Part
+    name = "SR_PLASMA_DISCHARGE"
+    description = "SR_FLAME_THROWER_DESC"
```
Comment (MatGB) [inline default/scripting/ship_parts/Weapons.focs.txt]: _DESC key not updated

**Comment by MatGB** [comment]
Oh wow, I love those new icons.

I'm going to merge this in now as my testing so far hasn't come up with anything you haven't just fixed, please do open a new PR for future work (and when I've finished testing this I'll think about whether the weapon stats are balanced enough, I think the Warden in particular needs stronger guns it used to have death ray 4)

**Comment by Vezzra** [comment]
@V-ktor: can you submit a PR to the assets repo for the graphics?

**Pull request merged**